kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1164 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/pointer_vector.h"
+#include "includes/variables.h"
+#include "includes/global_pointer_variables.h"
+#include "modified_shape_functions/triangle_2d_3_modified_shape_functions.h"
+#include "modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.h"
+#include "utilities/element_size_calculator.h"
+#include "utilities/mls_shape_functions_utility.h"
+#include "utilities/rbf_shape_functions_utility.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/shifted_boundary_meshless_interface_utility.h"
+
+namespace Kratos
+{
+
+namespace
+{
+    using GeometryType = ShiftedBoundaryMeshlessInterfaceUtility::GeometryType;

```

Comment:
could we pass the variable instead of using DISTANCE by default?

this goes in the direciton of using a different distance variable for example for free surface and for embedded, with the ultimate goal of being able to mix the two things

Reply:
Makes sense. Good observation.

Reply:
Done!